tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala

```diff
@@ -1814,4 +1814,31 @@ class ControlSyntaxSuite extends BaseDottySuite {
       )
     )
   }
+
+  test("if-infix") {
+    runTestAssert[Stat](
+      """|if (1) max 10 gt 0
+         |
```

Comment:
is this really valid in scala 3?!

Reply:
Yep, it's taken straight from the tests. I think this is the most hardcore edge case they could find.